lnrpc/routerrpc/router.proto

```diff
@@ -86,6 +86,17 @@ enum PaymentState {
     routes to the destination at all.
     */
     FAILED_NO_ROUTE = 3;
+
+    /**
+    A non-recoverable error has occured.
+    */
+    FAILED_ERROR = 4;
+
+    /**
+    Payment details incorrect (unknown hash, invalid amt or
+    invalid final cltv delta)
+    */
+    FAILED_INCORRECT_PAYMENT_DETAILS = 5;
```

Comment:
Will we also eventually surface this additional meta data? So the payment hash extended, amount, etc?

Reply:
We can't. Because of the anti-probe measure we took on the protocol level, we don't know which of the parameters is incorrect.